### PR TITLE
Reliable Topic Implementation

### PR DESCRIPTION
**Pull request by GhostInAMachine**

### CODE REVIEW

File: src/Address.ts [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+
+    readData(input: DataInput): any {
+        this.port = input.readInt();
+        // IPv4 or IPv6 - not used here
+        input.readByte();
+        this.host = input.readUTF();
+    }
+
+    writeData(output: DataOutput): void {
+        output.writeInt(this.port)
```
Comment (gAmUssA) [inline src/Address.ts]: @GhostInAMachine I  believe problem is here :)

**Comment by GhostInAMachine** [comment]
verify

**Comment by mustafaiman** [comment]
hey @GhostInAMachine . I see ReliableTopic requires a few more changes than we are used to. Can you add some description about why it is different than other proxy implementations?

**Comment by GhostInAMachine** [comment]
@mustafaiman 

- It has some client-side configuration, since it is a wrapper around a ringbuffer
- The topic message have references to address of the sender, so I had to make it serializable
- Topic message has some complex behavior, so its source code is split into separate files for clarity

**Comment by gAmUssA** [comment]
@mustafaiman are you good with merging this?

**Comment by mustafaiman** [comment]
verify

**Pull request merged**